Examples/SlackClone/SlackComposerVC.swift

```diff
@@ -94,10 +94,12 @@ open class ComposerEmojiSuggestionsDataSource: NSObject, UICollectionViewDataSou
     }
 
     public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(
+        guard let cell = collectionView.dequeueReusableCell(
             withReuseIdentifier: ComposerEmojiSuggestionsCollectionViewCell.reuseId,
             for: indexPath
-        ) as! ComposerEmojiSuggestionsCollectionViewCell
+        ) as? ComposerEmojiSuggestionsCollectionViewCell else {
+            return UICollectionViewCell()
+        }
```

Comment:
How do you feel about introducing `Reusable` type for both table & collection views? 🙂 The iOS community is familiar with this approach and for us, it would allow to remove the boilerplate and make code easy to read.

```swift
protocol Reusable {
    static var reuseIdentifier: String { get }
}

extension Reusable {
    static var reuseIdentifier: String {
        String(describing: self)
    }
}

extension UITableView {
    func register<Cell: UITableViewCell>(_ type: Cell.Type) where Cell: Reusable {
        register(type, forCellReuseIdentifier: type.reuseIdentifier)
    }
    
    func dequeueReusableCell<Cell: UITableViewCell>(_ type: Cell.Type, for indexPath: IndexPath) -> Cell where Cell: Reusable {
        // here it's acceptable to disable the force-case rule, i think
        return dequeueReusableCell(withIdentifier: type.reuseIdentifier, for: indexPath) as! Cell
    }
}
```